src/extractor/guidance/driveway_handler.cpp

```diff
@@ -0,0 +1,75 @@
+#include "extractor/guidance/driveway_handler.hpp"
+
+#include "util/assert.hpp"
+
+using osrm::extractor::guidance::getTurnDirection;
+using osrm::util::angularDeviation;
+
+namespace osrm
+{
+namespace extractor
+{
+namespace guidance
+{
+
+DrivewayHandler::DrivewayHandler(const IntersectionGenerator &intersection_generator,
+                                 const util::NodeBasedDynamicGraph &node_based_graph,
+                                 const std::vector<util::Coordinate> &coordinates,
+                                 const util::NameTable &name_table,
+                                 const SuffixTable &street_name_suffix_table)
+    : IntersectionHandler(node_based_graph,
+                          coordinates,
+                          name_table,
+                          street_name_suffix_table,
+                          intersection_generator)
+{
+}
+
+// The intersection without major roads needs to pass by service roads (bd, be)
+//       d
+//       .
+//  a--->b--->c
+//       .
+//       e
+bool DrivewayHandler::canProcess(const NodeID /*nid*/,
+                                 const EdgeID /*via_eid*/,
+                                 const Intersection &intersection) const
+{
+    const auto from_eid = intersection.front().eid;
+
+    if (intersection.size() <= 2 ||
+        node_based_graph.GetEdgeData(from_eid).road_classification.IsLowPriorityRoadClass())
+        return false;
```

Comment:
So this handler only handles situations where the main road is a high-priority road and there is a low-priority road crossing?